src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -793,20 +820,19 @@ def resolve_all_constraints(self) -> bool:
     def scratch_dir(self):
         return os.path.join(self.options.pants_workdir, *self.options_scope.split("."))
 
-    def compatibility_or_constraints(self, compatibility: Iterable[str] | None) -> tuple[str, ...]:
+    def compatibility_or_constraints(
+        self, compatibility: Iterable[str] | None, resolve: str | None
+    ) -> tuple[str, ...]:
         """Return either the given `compatibility` field or the global interpreter constraints.
 
         If interpreter constraints are supplied by the CLI flag, return those only.
         """
         if self.options.is_flagged("interpreter_constraints"):
             return self.interpreter_constraints
-        return tuple(compatibility or self.interpreter_constraints)
-
-    def compatibilities_or_constraints(
```

Comment:
Looks like this is not called anywhere?

Reply:
Yep, just confirming that this could have been removed all along, regardless of your change.